app/src/main/java/com/codesoom/assignment/interceptors/AuthenticationInterceptor.java

```diff
@@ -0,0 +1,68 @@
+package com.codesoom.assignment.interceptors;
+
+import com.codesoom.assignment.application.authentication.AuthenticationService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+import org.springframework.web.servlet.HandlerInterceptor;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@RequiredArgsConstructor
+@Component
+public class AuthenticationInterceptor implements HandlerInterceptor {
+    private final AuthenticationService authenticationService;
+
+    @Override
+    public boolean preHandle(
+            HttpServletRequest request,
+            HttpServletResponse response,
+            Object handler
+            ) throws Exception {
+        return filterWithPathAndMethod(request) ||
+                doAuthentication(request, response);
+    }
+
+    private boolean filterWithPathAndMethod(HttpServletRequest request) {
+        String path = request.getRequestURI();
+        String method = request.getMethod();
+        if (path.startsWith("/products") && (
+                method.equals("GET") || method.equals("OPTIONS"))
```

Comment:
`method`가 `null`인 경우에 여기에서 NPE가 터질 위험이 있습니다. 이렇게 수정해 봅시다.
```suggestion
                "GET".equals(method) || "OPTIONS".equals(method))
```

Reply:
NPE에 대한 생각을 전혀 하지 못했는데, 앞으로도 다시한번 생각해보겠습니다.

Reply:
ㅎㅎ 이 기법은 자바에서 흔한 패턴이니 꼭 기억해두세요.